context.go

```diff
@@ -37,11 +37,16 @@ func NewContext(app *App, set *flag.FlagSet, parentCtx *Context) *Context {
 
 	if c.Context == nil {
 		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			// wait for context cancellation
+			<-ctx.Done()
+			app.InterruptHandlerFunc(c)
+		}()
 		go func() {
 			defer cancel()
-			sigs := make(chan os.Signal, 1)
-			signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
-			<-sigs
+			signals := make(chan os.Signal, 1)
+			signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
+			<-signals
```

Comment:
I think this should be in a loop to deliver the signal more than once, and this should call the handler repeatedly.